### PR TITLE
feat: tekton for teams

### PR DESCRIPTION
**Pull request by j-zimnowoda**
implements #1298

Note: This PR requires extensive testing, including long run on QA cluster and team lifecycle.

### CODE REVIEW

File: helmfile.d/snippets/defaults.yaml
Comment (srodenhuis) [inline helmfile.d/snippets/defaults.yaml:1]: argocd is set to true, but when installing Otomi using Helm chart, argocd is not installed

**Comment by srodenhuis** [comment]
After installing, the first `otomi-tasks-teams` pipelineruns fail:
```
2023-12-10T12:57:18.226Z otomi:global:error Error: err: no releases found that matches specified selector() and environment(default), in any helmfile
    at hfCore (/home/app/stack/dist/src/common/hf.js:44:30)
    exit code: 3
    at ChildProcess.<anonymous> (/home/app/stack/node_modules/zx/dist/bundle.cjs:15261:22)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```
After creating the first team, the `otomi-tasks-teams` pipelineruns succeed.

**Comment by srodenhuis** [comment]
After I created the first team, Prometheus, Grafana and Alertmanager are installed while these apps are not enabled:
```
alertmanager-prometheus-test-kube-prome-alertmanager-0   3/3     Running    0          22m
hello-5fb5bd544c-52rc9                                   2/2     Running    0          6m46s
hello-5fb5bd544c-fnsw2                                   2/2     Running    0          6m46s
prometheus-prometheus-test-kube-prome-prometheus-0       3/3     Running    0          22m
prometheus-test-grafana-7f676f6dff-xjw7q                 4/4     Running    0          22m
prometheus-test-kube-prome-admission-patch-77dhh         1/2     NotReady   1          22m
prometheus-test-kube-prome-operator-55b58c7554-rflsg     2/2     Running    0          22m
prometheus-test-kube-state-metrics-5f5bcc8cd7-gg788      2/2     Running    0          22m
prometheus-test-prometheus-node-exporter-kxfhk           1/1     Running    0          22m
prometheus-test-prometheus-node-exporter-wkrtc           1/1     Running    0          22m
tekton-dashboard-59584c96b9-llzdn                        2/2     Running    0          22m
```
It looks like the team charts (`prometheus-{{ $teamId }}) are not used (see 60.teams)

**Comment by j-zimnowoda** [comment]
> After installing, the first `otomi-tasks-teams` pipelineruns fail:
> 
> ```
> 2023-12-10T12:57:18.226Z otomi:global:error Error: err: no releases found that matches specified selector() and environment(default), in any helmfile
>     at hfCore (/home/app/stack/dist/src/common/hf.js:44:30)
>     exit code: 3
>     at ChildProcess.<anonymous> (/home/app/stack/node_modules/zx/dist/bundle.cjs:15261:22)
>     at ChildProcess.emit (node:events:513:28)
>     at maybeClose (node:internal/child_process:1100:16)
>     at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
> ```
> 
> After creating the first team, the `otomi-tasks-teams` pipelineruns succeed.
Thanks a lot. I did not take into account that use case.

**Comment by j-zimnowoda** [comment]
> After I created the first team, Prometheus, Grafana and Alertmanager are installed while these apps are not enabled:
> 
> ```
> alertmanager-prometheus-test-kube-prome-alertmanager-0   3/3     Running    0          22m
> hello-5fb5bd544c-52rc9                                   2/2     Running    0          6m46s
> hello-5fb5bd544c-fnsw2                                   2/2     Running    0          6m46s
> prometheus-prometheus-test-kube-prome-prometheus-0       3/3     Running    0          22m
> prometheus-test-grafana-7f676f6dff-xjw7q                 4/4     Running    0          22m
> prometheus-test-kube-prome-admission-patch-77dhh         1/2     NotReady   1          22m
> prometheus-test-kube-prome-operator-55b58c7554-rflsg     2/2     Running    0          22m
> prometheus-test-kube-state-metrics-5f5bcc8cd7-gg788      2/2     Running    0          22m
> prometheus-test-prometheus-node-exporter-kxfhk           1/1     Running    0          22m
> prometheus-test-prometheus-node-exporter-wkrtc           1/1     Running    0          22m
> tekton-dashboard-59584c96b9-llzdn                        2/2     Running    0          22m
> ```
> 
> It looks like the team charts (`prometheus-{{ $teamId }}) are not used (see 60.teams)

Thanks a lot. I did not take into account that use case.

**Comment by j-zimnowoda** [comment]
> > After I created the first team, Prometheus, Grafana and Alertmanager are installed while these apps are not enabled:
> > ```
> > alertmanager-prometheus-test-kube-prome-alertmanager-0   3/3     Running    0          22m
> > hello-5fb5bd544c-52rc9                                   2/2     Running    0          6m46s
> > hello-5fb5bd544c-fnsw2                                   2/2     Running    0          6m46s
> > prometheus-prometheus-test-kube-prome-prometheus-0       3/3     Running    0          22m
> > prometheus-test-grafana-7f676f6dff-xjw7q                 4/4     Running    0          22m
> > prometheus-test-kube-prome-admission-patch-77dhh         1/2     NotReady   1          22m
> > prometheus-test-kube-prome-operator-55b58c7554-rflsg     2/2     Running    0          22m
> > prometheus-test-kube-state-metrics-5f5bcc8cd7-gg788      2/2     Running    0          22m
> > prometheus-test-prometheus-node-exporter-kxfhk           1/1     Running    0          22m
> > prometheus-test-prometheus-node-exporter-wkrtc           1/1     Running    0          22m
> > tekton-dashboard-59584c96b9-llzdn                        2/2     Running    0          22m
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It looks like the team charts (`prometheus-{{ $teamId }}) are not used (see 60.teams)
> 
> Thanks a lot. I did not take into account that use case.

Fixed

**Comment by srodenhuis** [comment]
I propose to add logic so `otomi-task` will only run when changes are made in values except in `/teams` folder. Note that the URL of a (new) team service is added to the REDIRECT_URIS of job-keycloak. But I do not think this is required because the option to add OICD (authentication) in front of a team service is not available (anymore) for users

**Comment by j-zimnowoda** [comment]
> note that the URL of a (new) team service is added to the REDIRECT_URIS of job-keycloak. 
Known issue: #685

**Comment by j-zimnowoda** [comment]
> I propose to add logic so `otomi-task` will only run when are made in values except in `/teams` folder.

It is not feasible right now, because the `apply` command is not aware of files that has been changed.

I would rather lean towards operators to solve this and many other issue with jobs.

**Review by srodenhuis — APPROVED** [review]

**Comment by j-zimnowoda** [comment]
There is an issue with Builds feature.
Since ArgoCD adds labels to PipelineRunTemplate the trigger will inherit the labels.
ArgoCD will recognise this as unwanted resource and kill it.

**Comment by j-zimnowoda** [comment]
> There is an issue with Builds feature.
> Since ArgoCD adds labels to PipelineRunTemplate the trigger will inherit the labels.
> ArgoCD will recognise this as unwanted resource and kill it.

The issues has been solved by setting the `resourceTrackingMethod` parameter

File: helmfile.d/snippets/defaults.yaml
Comment (j-zimnowoda) [reply helmfile.d/snippets/defaults.yaml:1]: hey, I was not able to reproduce it. But I modified the `.values/env/apps/argocd.yaml`  file that could case that issue.

**Pull request merged**